src/compiler/checker.ts

```diff
@@ -12149,6 +12185,9 @@ namespace ts {
                                 }
                             }
                         }
+                        if (prop.flags & SymbolFlags.SetAccessor) {

```

Comment:
I guess this check is insufficient if `prop` itself is a union or intersection property with a `writeType` derived from one or more set accessors unless we propagate this flag to the union/intersection property. I initially tried an approach where we propagate `Accessor` flags and it didn’t seem to mess anything up, but later I changed my approach for other reasons and it seemed that propagation was no longer necessary, not thinking of the nested unions/intersections case until I published the PR. The other option would be to check if this property does have a cached writeType or deferred write type similar to `getWriteTypeOfSymbol` above. Thoughts?

Reply:
Hm, I think we'd rather not expand how many places need to explicitly handle deferred Union calculation or other cached types; so it's probably better to propagate the flags if possible.

Reply:
> if `prop` itself is a union or intersection property

@weswigham uh dumb question, but any idea how to construct such a test? Union normalization is thwarting my very simple attempts.

Reply:
Ah, I think you might manage something by using the constraint of a conditional type to create a denormalized union or intersection? It's probably not particularly easy.

Reply:
In that case, maybe this check is fine?

Reply:
Couldn't we just always `getWriteTypeOfSymbol` (which should hopefully guarantee handling recursive processing of transient/deferred symbols) and assign the `writeType`-related fields any time they're different than the non-writing fields? Something like
```ts
const writeType = getWriteTypeOfSymbol(prop);
if (writeTypes || writeType !== type) {
  writeTypes = append(!writeTypes ? propTypes.slice() : writeTypes, writeType);
}
propTypes.push(type);
```
thus avoiding the upfront check for an accessor entirely?

Reply:
Yeah that works, thanks